### DISCUSSION TITLE
Compilation error on RHEL 8.4

### DISCUSSION THREAD

**Discussion by electrosaur** [abraunegg/onedrive#1504] [Q&A] [answered]
Using D compiler 2.097.0.

I get the following compilation error:
src/qxor.d(67): Error: need `this` for `lengthInBytes` of type `immutable(ulong)`

With v2.4.12 and master.

Any suggestions?
Thanks.

**Comment by abraunegg** [comment]
@electrosaur 
This is a compiler issue, due to 2.097.0 being released 2 days ago.

The work around is to use the prior release compiler:
```
curl -fsS https://dlang.org/install.sh | bash -s dmd-2.096.1
```
or to use the LDC compiler:
```
curl -fsS https://dlang.org/install.sh | bash -s ldc
```

Both these have been tested & work without issue

**Comment by abraunegg** [comment, marked as answer]
@electrosaur 
This has been resolved with PR #1505 which has been merged into 'master'
